### DISCUSSION TITLE
create a provider family from a mutable map?

### DISCUSSION THREAD

**Discussion by kylrth** [rrousselGit/riverpod#1342] [Q&A] [answered]
I have a map that I read from JSON at startup. I'd like to make the values in the map available as a provider family, and also allow mutating or adding values to the map in `StateNotifier` style. Is there a way to create a family of providers from a `StateNotifier`?

**Comment by rrousselGit** [comment, marked as answer]
You technically can pass mutable objects as family parameter. The important is that `==` stays the same (so in your case, no map cloning)

Although I would discourage it. Family parameters should be immutable. This way, when they update, the associated parameter will update too.
